packages/query-core/src/types.ts

```diff
@@ -407,6 +407,23 @@ export interface FetchQueryOptions<
   staleTime?: number
 }
 
+export interface EnsureQueryDataOptions<
+  TQueryFnData = unknown,
+  TError = DefaultError,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+  TPageParam = never,
+  > extends FetchQueryOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryKey,
+    TPageParam
+  > {
+  revalidateIfStale?: boolean
```

Comment:
going back and forth on my own proposal: since we don't use the word `revalidate` anywhere else, maybe it should be `refetchIfStale` ? Like `refetchOnMount` etc ... I think I like `revalidate` better, because it points towards it happening in the background, but I'm not sure if we should introduce new terminology 🤔

Reply:
Yes i agree. `revalidate` is not used anywhere but it indicates clearly that is will not just invalidate the data but also revalidate it. `fetchIfStale` indicates that it will fetch the data if stale so you need to wait on it. 

Imo `revalidateIfStale` describes best what happens but i agree about the terminology not being used elsewhere, But also this isn't a function used elsewhere expect on `useQuery` but there its default behaviour.

Reply:
> but there its default behaviour.

yes but you can turn it off with `refetchOnMount: false` for example. So maybe we should stick to `refetchIfStale` ? and then maybe rename everything to `revalidate`, so `revalidateOnMount` etc. in the next major ?

or would it make more sense to start with `revalidateIfStale` here as a new API and then streamline it in the next major. I think I like this option more